pkg/virt-operator/util/config.go

```diff
@@ -526,3 +535,37 @@ func IsValidLabel(label string) bool {
 	r := regexp.MustCompile(`^([a-z0-9A-Z]([a-z0-9A-Z\-\_\.]{0,61}[a-z0-9A-Z])?)?$`)
 	return r.Match([]byte(label))
 }
+
+// GetGsInfo returns the libguestfs-tools image information based on the KubeVirt installation in the namespace.
+func GetGsInfo(virtCli kubecli.KubevirtClient, namespace string) func(_ *restful.Request, response *restful.Response) {
+	return func(_ *restful.Request, response *restful.Response) {
+		var kvConfig KubeVirtDeploymentConfig
+		kvlist, err := virtCli.KubeVirt(namespace).List(&metav1.ListOptions{})
```

Comment:
We should avoid making an API call to k8s here, especially since this API involves no auth.

All the information should be available though, we just need to pass it into the right place.

virt-api already has a KubeVirt object informer. Pass the `kubeVirtInformer` in pkg/virt-api/api.go into GetGsInfo somehow. You might need to store the `KubeVirtInformer` object on the virtAPIApp struct in order to retrieve the informer at the time that the route is created. There's some indirection going on, but it shoudl be possible.

Reply:
@davidvossel many thanks for the feedback!
Actually, it already has the informer in the [configCluster](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-api/api.go#L101) variable and the function [getConfigFromKubeVirtCR](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-config/config-map.go#L523) does exactly what I need.
I exposed it and use it in the virt-api. I updated the PR accordingly (commit: 08d0cd1815b925e1f904df137cf14fa455ff2d6e). Please, let me know if this works better

Reply:
:+1:  yup definitely, thanks!